lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -0,0 +1,451 @@
+        """Removes the vault CA certificate secret in the peer relation.
+
+        Only the key from the relation is removed. The actual juju secret is unmodified.
+        """
+        if not self.charm._is_peer_relation_created():
+            raise RuntimeError("Peer relation not created")
+        peer_relation = self.model.get_relation(self.peer_relation)
+        if peer_relation:
+            peer_relation.data[self.charm.app].pop(CA_CERTIFICATE_JUJU_SECRET_KEY)
+            logger.debug("Vault CA certificate secret removed from peer relation")
+
+    def _ca_certificate_set_in_peer_relation(self) -> bool:
+        """Returns whether CA certificate is stored in peer relation data."""
+        try:
+            ca_private_key, ca_certificate = self._get_ca_certificate_secret_in_peer_relation()
+            if ca_private_key and ca_certificate:
+                return True
+        except PeerSecretError:
+            return False
+        return False
+
+    """TODO: The following are methods that go through the container.
+    They will be extracted out in the future."""
+
+    def _remove_all_certs_from_workload(self) -> None:
+        """Removes the certificate files that are used for authentication."""
+        self._remove_tls_file_from_workload(File.CA)
+        self._remove_tls_file_from_workload(File.CERT)
+        self._remove_tls_file_from_workload(File.CSR)
+        logger.debug("Removed existing certificate files from workload.")
+
+    def _reload_vault_container(self) -> None:
+        """Sends a SIGHUP signal to the container which reloads Vault's files. Fails gracefully."""
+        try:
+            self.charm._container.send_signal(SIGHUP, self.charm._container_name)
+            logger.debug("Container restart requested")
+        except APIError:
+            logger.debug("Couldn't send signal to container. Proceeding normally.")
+
+    def _pull_tls_file_from_workload(self, file: File) -> str:
+        """Get a file related to certs from the container.
+
+        Args:
+            file: a File object that determines which file to read.
+
+        Returns:
+            str: The file content without whitespace.
+        """
+        try:
+            file_content: TextIO = self.charm._container.pull(
+                f"{TLS_FILE_FOLDER_PATH}/{file.name.lower()}.pem",
+            )
+        except PathError:
+            return ""
+        return file_content.read().strip()
+
+    def _push_tls_file_to_workload(self, file: File, data: str) -> None:
+        """Push one of the given file types to the workload.
+
+        Args:
+            file: a File object that determines which file to write.
+            data: the data to write into that file.
+        """
+        self.charm._container.push(
+            path=f"{TLS_FILE_FOLDER_PATH}/{file.name.lower()}.pem", source=data
+        )
+        logger.debug("TLS: Pushed %s file to workload", file.name)
```

Comment:
I don't know that it is necessary to add the prefix to each log line? Won't the specific python file be present in the log itself? Probably worth validating

Reply:
`unit-vault-k8s-0: 12:58:54 INFO unit.vault-k8s/0.juju-log TLS:`

Reply:
I found the way, and added an adapter at the top of the file that will prepend `[TLS]` to all logging done in this file.